homeassistant/components/sensor/foobot.py

```diff
@@ -0,0 +1,151 @@
+        for device in devices:
+            for sensor_type in SENSOR_TYPES.keys():
+                if sensor_type == 'time':
+                    continue
+                foobot_sensor = FoobotSensor(client, data, device,
+                                             sensor_type)
+                dev.append(foobot_sensor)
+    except (aiohttp.client_exceptions.ClientConnectorError,
+            asyncio.TimeoutError, FoobotClient.TooManyRequests,
+            FoobotClient.InternalError):
+        _LOGGER.exception('Failed to connect to foobot servers.')
+        raise PlatformNotReady
+    except (FoobotClient.ClientError):
+        _LOGGER.exception('Failed to fetch data from foobot servers.')
+        raise HomeAssistantError
+    async_add_devices(dev, True)
+
+
+class FoobotSensor(Entity):
+    """Implementation of a Foobot sensor."""
+
+    __cache = {}
+
+    def __init__(self, client, data, device, sensor_type):
+        """Initialize the sensor."""
+        self._client = client
+        self._uuid = device['uuid']
+        self._name = 'Foobot {} {}'.format(device['name'],
+                                           SENSOR_TYPES[sensor_type][0])
+        self.type = sensor_type
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        try:
+            data = self.__cache[self._uuid][self.type]
+        except(KeyError, TypeError):
+            data = None
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity."""
+        return self._unit_of_measurement
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data."""
+        if self._uuid in self.__cache:
+            last_request = self.__cache[self._uuid]['last_request']
+            _LOGGER.debug('last_update: %s', last_request)
+            _LOGGER.debug('now: %s', datetime.utcnow())
+            if last_request + SCAN_INTERVAL > datetime.utcnow():
+                # If the last request made is recent enough,
+                # just skip the update
+                _LOGGER.debug('Skipping update')
+                return
+
+        interval = SCAN_INTERVAL.total_seconds()
+        try:
+            response = yield from self._client.get_last_data(self._uuid,
+                                                             interval,
+                                                             interval + 1)
+        except (aiohttp.client_exceptions.ClientConnectorError,
+                asyncio.TimeoutError, self._client.TooManyRequests,
+                self._client.InternalError):
+            raise HomeAssistantError
```

Comment:
There is no use in raising here. You can log an error to the user and just not do anything. The error will just fill up the log.

Also, when you get `TooManyRequests`, would you want to do a backoff algorithm?